lms/djangoapps/instructor_task/docs/decisions/002-processing-scheduled-instructor-tasks.rst

```diff
@@ -0,0 +1,31 @@
+=====================================
+Processing Scheduled Instructor Tasks
+=====================================
+
+Status
+------
+
+Draft
+
+Background
+----------
+The previous ADR (``Scheduled Instructor Tasks``) explains the motivation and approach for scheduling instructor tasks. 
+
+This ADR will cover our approach for the execution of scheduled instructor tasks.
+
+Decision
+--------
+
+A management command will be introduced to assist in processing scheduled instructor tasks. This management command will rely on new utility functions in the **instructor_task** app of edx-platform to query tasks that have the **SCHEDULED** status and then determine if these tasks are due for execution.
+
+This management command will be invoked by a Jenkins job running on a (cron) schedule to process scheduled tasks due for execution.	The job will run every fifteen minutes.
```

Comment:
This sounds like it has business and UI implications. Does this mean instructors / course teams can only schedule emails on quarter-hours? (ie. 12:00, 12:15, 12:30, 1:00)? Is Product okay with this?

Reply:
If the command is lightweight enough in processing time and DB row-locking, perhaps it can run at shorter intervals?
Is there any protection needed against multiple mgmt commands running at once? Will that condition be handled by row-locking at the DB level? The main worry is a scheduled task being executed multiple times.
If the Jenkins job fails, it should fail loudly as it'll prevent any scheduled tasks from being queued during the failure interval.